sqlglot/dialects/doris.py

```diff
@@ -0,0 +1,111 @@
+from __future__ import annotations
+
+import typing as t
+
+from sqlglot import exp, generator
+from sqlglot.dialects.dialect import (
+    approx_count_distinct_sql,
+    arrow_json_extract_sql,
+    rename_func,
+)
+from sqlglot.dialects.mysql import MySQL
+from sqlglot.helper import seq_get
+
+
+def _to_date_sql(self: generator.Generator, expression: exp.TsOrDsToDate) -> str:
+    this = self.sql(expression, "this")
+    self.format_time(expression)
+    return f"TO_DATE({this})"
+
+
+def _time_format(
+    self: generator.Generator, expression: exp.UnixToStr | exp.StrToUnix
+) -> t.Optional[str]:
+    time_format = self.format_time(expression)
+    if time_format == Doris.TIME_FORMAT:
+        return None
+    return time_format
+
+
+class Doris(MySQL):
+    DATE_FORMAT = "'yyyy-MM-dd'"
+    DATEINT_FORMAT = "'yyyyMMdd'"
+    TIME_FORMAT = "'yyyy-MM-dd HH:mm:ss'"
+
+    TIME_MAPPING = {
+        "%M": "%B",
+        "%m": "%%-M",
+        "%c": "%-m",
+        "%e": "%-d",
+        "%h": "%I",
+        "%S": "%S",
+        "%u": "%W",
+        "%k": "%-H",
+        "%l": "%-I",
+        "%W": "%a",
+        "%Y": "%Y",
+        "%d": "%%-d",
+        "%H": "%%-H",
+        "%s": "%%-S",
+        "%D": "%%-j",
+        "%a": "%%p",
+        "%y": "%%Y",
+        "%": "%%",
+    }
+
+    class Parser(MySQL.Parser):
+        FUNCTIONS = {
+            **MySQL.Parser.FUNCTIONS,
+            "DATE_TRUNC": lambda args: exp.TimestampTrunc(
+                this=seq_get(args, 1), unit=seq_get(args, 0)
+            ),
+        }
+
+    class Generator(MySQL.Generator):
+        CAST_MAPPING = {}
+
+        TYPE_MAPPING = {
+            **MySQL.Generator.TYPE_MAPPING,
+            exp.DataType.Type.TEXT: "STRING",
+            exp.DataType.Type.TIMESTAMP: "DATETIME",
+            exp.DataType.Type.TIMESTAMPTZ: "DATETIME",
+        }
+
+        TRANSFORMS = {
+            **MySQL.Generator.TRANSFORMS,
+            exp.ApproxDistinct: approx_count_distinct_sql,
+            exp.JSONExtractScalar: arrow_json_extract_sql,
+            exp.JSONExtract: arrow_json_extract_sql,
+            exp.DateDiff: rename_func("DATEDIFF"),
+            exp.RegexpLike: rename_func("REGEXP"),
```

Comment:
Need to add an entry in `Parser.FUNCTIONS` for this:

```python
"REGEXP": exp.RegexpLike.from_arg_list
```

Ditto about Starrocks suffering from the same issue.

Reply:
OK, thanks